apps/web/app/api/v1/management/responses/route.ts

```diff
@@ -2,17 +2,25 @@ import { authenticateRequest } from "@/app/api/v1/auth";
 import { responses } from "@/app/lib/api/response";
 import { NextRequest } from "next/server";
 
-import { getResponsesByEnvironmentId } from "@formbricks/lib/response/service";
+import { getResponses, getResponsesByEnvironmentId } from "@formbricks/lib/response/service";
 import { DatabaseError } from "@formbricks/types/errors";
+import { TResponse } from "@formbricks/types/responses";
 
 export const GET = async (request: NextRequest) => {
-  const surveyId = request.nextUrl.searchParams.get("surveyId");
+  const searchParams = request.nextUrl.searchParams;
+  const surveyId = searchParams.get("surveyId");
+  const limit = searchParams.get("limit") ? Number(searchParams.get("limit")) : undefined;
+  const offset = searchParams.get("skip") ? Number(searchParams.get("skip")) : undefined;
+
   try {
     const authentication = await authenticateRequest(request);
     if (!authentication) return responses.notAuthenticatedResponse();
-    let environmentResponses = await getResponsesByEnvironmentId(authentication.environmentId!);
+    let environmentResponses: TResponse[] = [];
+
     if (surveyId) {
-      environmentResponses = environmentResponses.filter((response) => response.surveyId === surveyId);
+      environmentResponses = await getResponses(surveyId, limit, offset);
+    } else {
+      environmentResponses = await getResponsesByEnvironmentId(authentication.environmentId!, limit, offset);
```

Comment:
explicitly saying that a value exists using `!` is normally a warning sign. Would it make sense to add a check first to make sure that the value of `authentication.environmentId` exists?